### DISCUSSION TITLE
Vim

### DISCUSSION THREAD

**Discussion by KwatME** [fonsp/Pluto.jl#2123] [Questions about using Pluto]
Thank you for this amazing project.
Is there a way to use vim binding with `Pluto.jl`?

**Comment by icweaver** [comment]
fwiw, I think using Pluto's new [autoreload feature](https://github.com/fonsp/Pluto.jl/pull/1029) combined with vim/neovim's [`autoread`](https://neovim.io/doc/user/options.html#'autoread') (I think it's on by default in neovim?) is pretty slick:

https://user-images.githubusercontent.com/25312320/169415733-20af9b4c-b374-46c3-bd5f-ef13deeb3ae0.mp4

I just add something like this to my init.vim (assuming vim)

```vim
set autoread
set updatetime=100
autocmd CursorHold * checktime | call feedkeys("lh")
```

or this to my init.lua (assuming neovim)

```lua
-- Aliases
local opt = vim.opt
local cmd = vim.cmd

opt.updatetime = 100
cmd[[au CursorHold * checktime | call feedkeys("lh")]]
```

to get the desired behavior when the buffer is not focused. More discussion here: https://vi.stackexchange.com/a/14833/36984

I think `updatetime` is around 4000 milliseconds by default, so I just lowered it to make things feel more snappy

**Reply by holomorphism** [reply]
In addition to the settings described by @icweaver, I have added the following:

1. install julia-vim (https://github.com/JuliaEditorSupport/julia-vim).
2. put the following in ~/.vim/after/syntax/julia.vim:
   ```vim
   " fold cell order lines
   syntax region plutoCellOrder
         \ transparent
         \ start="\%(^# ╔═╡ Cell order:$\)\@="
         \ end="\%$"
         \ contains=juliaCommentL
         \ fold

   " fold multiline strings
   syntax region juliaString
         \ matchgroup=juliaStringDelim
         \ start=+\z("\(""\)\?\)+
         \ skip=+\%(\\\\\)*\\"+
         \ end=+\z1+
         \ contains=@juliaStringVars,@juliaSpecialChars,@juliaSpellcheckStrings
         \ fold
   syntax region juliaStringPrefixed
         \ contained
         \ matchgroup=juliaStringDelim
         \ start=+[^{([:space:]<>"]\+\z("\(""\)\?\)+
         \ skip=+\%(\\\\\)*\\"+
         \ end=+\z1+
         \ contains=@juliaSpecialCharsRaw
         \ fold
   ```

In Pluto julia files, `PLUTO_MANIFEST_TOML_CONTENTS` etc. can be very long, which is inconvenient when editing the julia file directly.
I am folding Markdown cells and Cell order lines with the above settings.
![](https://user-images.githubusercontent.com/3076585/169632749-a145ff84-eede-492d-a3ed-ca641463011a.gif)

**Reply by KwatME** [reply]
Thank you all. Is there a way to add a new cell from editor instead of clicking Pluto?

**Reply by icweaver** [reply]
I could be mistaken about which plugin @rikhuijzer is referring to, but I think this one might have support for it. Haven't tried it myself though https://github.com/hasundue/vim-pluto

**Comment by rikhuijzer** [comment]
> Is there a way to use vim binding with `Pluto.jl`?

I'm a huge fan of Vim myself and use it all the time. For Pluto though I make an exception and stick to the default shortcuts which are mostly similar to Visual Studio Code. See https://github.com/fonsp/Pluto.jl/issues/65 for more info.

There also was once a Vim plugin for Pluto if I remember correctly. It was mentioned somewhere on Discourse. Cannot find it back th
